torch/csrc/jit/passes/quantization.cpp

```diff
@@ -692,6 +695,140 @@ c10::optional<IValue> toTwoElementIntList(Value* v) {
+    for (auto& method : module_.get_methods()) {
+      findModuleUses(method.name());
+    }
+
+    for (auto& method : module_.get_methods()) {
+      dedupModuleUses(method.name());
+    }
+  }
+
+  // Analyze the code to record necessary information
+  // to dedup the module uses
+  void findModuleUses(const std::string& method_name) {
+    script::Method method = module_.get_method(method_name);
+    auto graph = method.graph();
+
+    std::stack<Block*> blocks_to_visit;
+    blocks_to_visit.push(graph->block());
+    Value* self = graph->inputs()[0];
+    while (!blocks_to_visit.empty()) {
+      Block* b = blocks_to_visit.top();
+      blocks_to_visit.pop();
+      for (Node* n : b->nodes()) {
+        for (Block* subblock : n->blocks()) {
+          blocks_to_visit.push(subblock);
+        }
+        if (n->kind() != prim::CallMethod) {
+          continue;
+        }
+        std::vector<std::string> path;
+        Value* instance = n->inputs()[0];
+        // Iterator to traverse back the GetAttr calls
+        Value* iter = instance;
+        // trace back the instance to recover the path of the submodule
+        while (!hitGraphInput(iter)) {
+          Node* get_attr = iter->node();
+          // record the name of GetAttr
+          path.push_back(get_attr->s(attr::name));
+          // trace back the chain of GetAttr
+          iter = get_attr->inputs()[0];
+        }
+
+        if (iter == self) {
+          value_to_path_map_[instance] = path;
+          auto m = findChildModule(module_, path);
+          if (module_set_.find(m._ivalue()) == module_set_.end()) {
+            module_set_.insert(m._ivalue());
+          } else {
+            use_to_rewrite_.push_back(instance);
+          }
+        } else {
+          GRAPH_DEBUG(
+              "Can't handle the case of calling GetAttr on input ",
+              " of the graph in make submodule uses unique");
+        }
+      }
+    }
+  }
+
+  // Deduplicate module uses given the information we recorded before
+  void dedupModuleUses(const std::string& method_name) {
+    for (Value* v : use_to_rewrite_) {
+      const auto& path = value_to_path_map_[v];
+      const auto& m = findChildModule(module_, path);
+      // add a clone of the child module to the parent of the duplicated module
+      const auto& child_name = addChildModule(module_, m, path);
+      TORCH_INTERNAL_ASSERT(v->node()->kind() == prim::GetAttr);
+      // change the name in GetAttr call
+      auto original_name = v->node()->s(attr::name);
+      v->node()->s_(attr::name, child_name);
+      GRAPH_DEBUG(
+          "Module use dedup: changing original module ",
+          original_name,
+          " to ",
+          child_name);
+    }
+  }
+
+ private:
+  script::Module module_;
+  // Map from value of module instance to the list of names of submodules
+  // starting from the top level module, e.g. ["sub1", "sub2", "relu"]
+  std::unordered_map<Value*, std::vector<std::string>> value_to_path_map_;
```

Comment:
This might be a commonly useful feature if we generalize it a bit. E.g. we can add an API that would take a module and return you a map from `ivalue::object` to `std::vector<std::string>` representing a path to each submodule.

Reply:
that sounds a little bit different from what we have here, here we have a `Value` in IR and traced back in the graph to get the path and then from path we get the `ivalue::Object`. Maybe we can think about generalize this when similar use cases comes up later

Reply:
With the current exact implementation yes, it's not general enough. However, locating a child module by its path from the root module seems general and I think it would've been useful here as well.

Reply:
Also, might be worth mentioning that this map is just a cache for `getModuleAccessPath` calls.

Reply:
we do have this function in the PR :) `findChlidModule`, although this function takes a list, are you talking about an API to get the child module given a string path like "a.b.c"? maybe we can have this later if there is a need